src/comlink.ts

```diff
@@ -281,11 +281,34 @@ export const transferHandlers = new Map<
   ["throw", throwTransferHandler],
 ]);
 
-export function expose(obj: any, ep: Endpoint = self as any) {
+function isAllowedOrigin(
+  origins: (string | RegExp)[],
+  origin: string
+): boolean {
+  for (const allowedOrigin of origins) {
+    if (origin === allowedOrigin || allowedOrigin === "*") {
+      return true;
+    }
+    if (allowedOrigin instanceof RegExp && allowedOrigin.test(origin)) {
+      return true;
+    }
+  }
+  return false;
+}
+
+export function expose(
+  obj: any,
+  ep: Endpoint = self as any,
+  origins: (string | RegExp)[] = ["*"]
+) {
   ep.addEventListener("message", function callback(ev: MessageEvent) {
     if (!ev || !ev.data) {
       return;
     }
+    if (!isAllowedOrigin(origins, ev.origin)) {
+      console.warn(`Invalid origin '${ev.origin}' for comlink proxy`);
```

Comment:
Not sure exactly how to handle invalid messages, a warning seems useful, throwing an exception can be pretty hard to handle in this case but also seems more correct. Thoughts @surma ?

Reply:
Yeah, I agree with a warning.